.ci/scripts/test_old_deps.sh

```diff
@@ -9,12 +12,56 @@ set -ex
 
 apt-get update
 apt-get install -y \
-        python3 python3-dev python3-pip python3-venv \
-        libxml2-dev libxslt-dev xmlsec1 zlib1g-dev tox libjpeg-dev libwebp-dev
+        python3 python3-dev python3-pip python3-venv pipx \
+        libxml2-dev libxslt-dev xmlsec1 zlib1g-dev libjpeg-dev libwebp-dev
 
 export LANG="C.UTF-8"
 
 # Prevent virtualenv from auto-updating pip to an incompatible version
 export VIRTUALENV_NO_DOWNLOAD=1
 
-exec tox -e py3-old
+# I'd prefer to use something like this
+#   https://github.com/python-poetry/poetry/issues/3527
+#   https://github.com/pypa/pip/issues/8085
+# rather than this sed script. But that's an Opinion.
+
+# patch the project definitions in-place
+# replace all lower bounds with exact bounds
+# but make the pyopenssl 17.0, which can work against an
+# OpenSSL 1.1 compiled cryptography (as older ones don't compile on Travis).
+# delete all lines referring to psycopg2 --- so no testing of postgres support
+# Omit systemd: we're not logging to journal here.
+
+sed -i-backup \
+   -e "s/[~>]=/==/g" \
```

Comment:
This character class should include a caret to account for caret requirements: https://python-poetry.org/docs/dependency-specification#caret-requirements

Reply:
I left a TODO for this in e96fd31ef.